allennlp/training/trainer.py

```diff
@@ -584,21 +584,22 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
 
             batch_group_outputs = []
             for batch in batch_group:
-                with amp.autocast(self._use_amp):
-                    batch_outputs = self.batch_outputs(batch, for_training=True)
-                    batch_group_outputs.append(batch_outputs)
-                    loss = batch_outputs.get("loss")
-                    reg_loss = batch_outputs.get("reg_loss")
-                    if torch.isnan(loss):
-                        raise ValueError("nan loss encountered")
-                    loss = loss / len(batch_group)
-
-                    batch_loss = loss.item()
-                    train_loss += batch_loss
-                    if reg_loss is not None:
-                        reg_loss = reg_loss / len(batch_group)
-                        batch_reg_loss = reg_loss.item()
-                        train_reg_loss += batch_reg_loss
+                # TODO(mattg): vision stuff is incompatible with 1.6, removing autocast for now.
+                # THIS SHOULD NOT BE COMMITTED TO MASTER!
```

Comment:
@matt-gardner, what is this? The detectron modules fail when we use `amp.autocast`?